src/libs/actions/EmojiPickerAction.ts

```diff
@@ -0,0 +1,77 @@
+import {ValueOf} from 'type-fest';
+import React from 'react';
+import {View} from 'react-native';
+import CONST from '../../CONST';
+
+type AnchorOrigin = {
+    horizontal: ValueOf<typeof CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL>;
+    vertical: ValueOf<typeof CONST.MODAL.ANCHOR_ORIGIN_VERTICAL>;
+};
+
+type EmojiPicker = {
+    showEmojiPicker: (onModalHideValue?: () => void, onEmojiSelectedValue?: () => void, emojiPopoverAnchor?: View, anchorOrigin?: AnchorOrigin, onWillShow?: () => void, id?: string) => void;
+    isActive: (id: string) => boolean;
+    hideEmojiPicker: (isNavigating: boolean) => void;
+    isEmojiPickerVisible: boolean;
+    resetEmojiPopoverAnchor: () => void;
+};
+
+const emojiPickerRef = React.createRef<EmojiPicker>();
+
+/**
+ * Show the EmojiPicker modal popover.
+ *
+ * @param onModalHide - Run a callback when Modal hides.
+ * @param onEmojiSelected - Run a callback when Emoji selected.
+ * @param emojiPopoverAnchor - Element on which EmojiPicker is anchored
+ * @param anchorOrigin - Anchor origin for Popover
+ * @param onWillShow - Run a callback when Popover will show
+ * @param id - Unique id for EmojiPicker
+ */
+function showEmojiPicker(onModalHide = () => {}, onEmojiSelected = () => {}, emojiPopoverAnchor = undefined, anchorOrigin = undefined, onWillShow = () => {}, id = undefined): void {
```

Comment:
```suggestion
function showEmojiPicker(onModalHide = () => {}, onEmojiSelected = () => {}, emojiPopoverAnchor = undefined, anchorOrigin = undefined, onWillShow = () => {}, id = undefined) {
```

Reply:
@blazejkustra @fabioh8010 did we have a style guide on whether to add `void` to a function returning undefined?

Reply:
We have a section that says we should rely on inferred type in such occasions, I'll change it.